x/ecocredit/server/core/query_classes.go

```diff
@@ -0,0 +1,41 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// Classes queries for all credit classes with pagination.
+func (k Keeper) Classes(ctx context.Context, request *v1beta1.QueryClassesRequest) (*v1beta1.QueryClassesResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
```

Comment:
I think we should move `x/ecocredit/server/core/query_utils.go` to global scope `types` or `app/utils`. It smells generic.